python/pyarrow/lib.pyx

```diff
@@ -21,8 +21,10 @@
 
 import datetime
 import decimal as _pydecimal
+HAS_NUMPY=False
 try:
     import numpy as np
+    HAS_NUMPY=True
 except ImportError:
     np = None
```

Comment:
What's the point of this? You can already write  `np is not None`. You don't have to define another internal variable.

Reply:
That was my suggestion to have a `HAS_NUMPY` or similar variable. This can then be used in other pyx files, I personally find that a bit easier to read than `np is not None` (but if in practice it turns out this is hardly used, might not be worth defining)

Reply:
`HAS_NUMPY` is another internal variable that people have to remember, so I'm skeptical about it.

`<module name> is [not] None` is a widespread idiom for optional dependencies.

Reply:
To be clear I don't feel strongly about it, `np is not None` will also work just fine (although easy to understand as well, it's just not an idiom I am familiar with from other projects)